### PR TITLE
feat: Sign terms

### PR DESCRIPTION
**Pull request by 0xonramp**
Adds flow to sign terms.

### CODE REVIEW

File: src/lib/providers/sign-terms-provider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  onCloseSignTermsModal: () => {},
+  setHasSignedTerms: () => {},
+  signTermsOfService: () => {},
+})
+
+interface Props {
+  children: ReactNode
+}
+
+const SIGN_TERMS_KEY = 'termsOfService'
```
Comment (0xonramp) [inline src/lib/providers/sign-terms-provider/index.tsx:37]: Can support other 'types' of confirmations in the future if needed.

File: src/app/providers.tsx [hunk 2]
```diff
@@ -26,14 +27,16 @@ export function Providers({ children }: { children: React.ReactNode }) {
...
+          <SignTermsProvider>
+            <QueryClientProvider client={queryClient}>
+              <RainbowKitProvider
+                theme={rainbowkitTheme}
+                appInfo={rainbowKitAppInfo}
+              >
+                <AnalyticsProvider>{children}</AnalyticsProvider>
+              </RainbowKitProvider>
+            </QueryClientProvider>
+          </SignTermsProvider>
```
Comment (janndriessen) [inline src/app/providers.tsx:39]: Wouldn't it be better to place this inside QueryClient and RainbowKit? So just wrapping the AnalyticsProvider.

Comment (janndriessen) [reply src/app/providers.tsx:39]: Or maybe even inside that one in case you wanna use logic from there.

File: src/lib/utils/api/index-api.ts [hunk 3]
```diff
@@ -18,4 +18,19 @@ export class IndexApi {
       throw error
     }
   }
+
+  async put(path: string, data: Record<string, unknown>) {
+    try {
+      const resp = await fetch(`${IndexApiBaseUrl}${path}`, {
+        method: 'PUT',
```
Comment (janndriessen) [inline src/lib/utils/api/index-api.ts:25]: Should probably check if would always just want one record or need a history of records when the user signed e.g. updated terms.

File: src/lib/providers/sign-terms-provider/index.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    if (!address) return
+    fetchSignature()
+  }, [address])
+
+  const signTermsOfService = async () => {
+    try {
+      const indexApi = new IndexApi()
+      await indexApi.put(`/signature/${address}`, {
+        [SIGN_TERMS_KEY]: new Date().toISOString(),
+      })
```
Comment (janndriessen) [inline src/lib/providers/sign-terms-provider/index.tsx:71]: I'd prefer to start with more fields right away. Let's talk about it offline.

File: src/app/providers.tsx (hunk 2 above)
Comment (0xonramp) [reply src/app/providers.tsx]: @janndriessen good call - done 👍

File: src/components/swap/index.tsx [hunk 5]
```diff
@@ -248,7 +247,7 @@ export const Swap = (props: SwapProps) => {
     }
 
     if (buttonState === TradeButtonState.signTerms) {
-      onOpenSignTermsModal()
+      await signTermsOfService()
```
Comment (janndriessen) [inline src/components/swap/index.tsx:250]: probably want a `return` just to be sure

Comment (0xonramp) [reply src/components/swap/index.tsx:250]: good call, will add 👍

**Pull request merged**